packages/core/src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -289,7 +290,10 @@ export async function uploadArtifactToS3(
         getLogger().error(
             `Amazon Q is unable to upload workspace artifacts to Amazon S3 for security scans. For more information, see the Amazon Q documentation or contact your network or organization administrator.`
         )
-        const errorMessage = (error as Error).message
+        const errorMessage = getTelemetryReasonDesc(error)?.includes(`"PUT" request failed with code "403"`)
+            ? `UploadArtifactToS3Exception: "PUT" request failed with code "403"`
```

Comment:
Don't we need add the backend errors here as well just like JetBrains. like 407s etc.., event though they're not seen here, they can happen right.

Reply:
Yes they may happen but we dont know the error message etc so didn't handle it. Thanks to Justin, by using this approach of `reasonDesc` as we can diff the error from where they are originating from.